### PR TITLE
Asset logs list

### LINKED ISSUE #2985: Component for asset logs
Part of epic #2896 

## Is your feature request related to a problem? Please describe 👀
[Figma designs](https://www.figma.com/file/rIK2zVFUVhtbuXo0RBzOzw/0033-2305_omSupply-ColdChain-UI?type=design&node-id=2653-16424&mode=design&t=8KRUuAo7s1V0r0RM-0)

image

image

## Describe the solution you'd like 🎁

Asset logs: TBD
CCEI logs: [See list here](https://docs.google.com/spreadsheets/d/1l1x38KT5cdt4Zd5piMWp3FuBuiSUv4faMHP8utDJDqU/edit#gid=505652337)

**Navigation:** 
1. Home > Cold Chain > Asset Register: List View > Click on an asset > Asset details view > Click on tab “Logs” > Click on add log entry
  Home > Cold Chain > CCE Register: List View > Click on a CCE > CCE: Details View > Click on tab “Logs” > Click on add log entry
2. Home > Cold Chain > CCE Register list view > Click on an asset > Asset details view > Click on “Update Status”
  Home > Cold Chain > CCE Register list view > Click on a CCE > CCE: Details View > Click on “Update Status”

**Permissions:** 
only users with the right permissions can create a new log entry

1. User selects a new functional status from a dropdown list (functional status could stay the same, for example, when the technician cannot repair it)
2. User selects a reason for the new functional status
3. User selects a type of log from a dropdown
    1. Malfunction Report
    2. Maintenance/Repair Service
    3. Planned Preventive Maintenance
4. User enters a comments (quite large)
5. User can upload photos that goes with the report
6. User clicks on “Save” to finalise the log

### Moneyworks Jobcode 🧰

`OMSCC:AST`

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2985 

# 👩🏻‍💻 What does this PR do? 
 
Fills out the `Status Logs` tab, and also adds the `Log` tab for the activity log.

Screenshot 2024-03-14 at 11 40 29 AM

Screenshot 2024-03-14 at 5 21 13 PM

Screenshot 2024-03-14 at 5 21 02 PM

# 🧪 How has/should this change been tested? 

Create some status logs and see if they are visible.

## 📃 Documentation

Part of the asset / CCEI feature

### CODE REVIEW

File: client/packages/coldchain/src/Equipment/DetailView/DetailView.tsx [hunk 1]
```diff
@@ -72,13 +75,21 @@ export const EquipmentDetailView = () => {
       value: 'Summary',
     },
     {
-      Component: <StatusLogs />,
+      Component: draft === undefined ? null : <StatusLogs assetId={draft.id} />,
```
Comment (mark-prins) [inline client/packages/coldchain/src/Equipment/DetailView/DetailView.tsx:78]: to ensure that the assetId is provided - just makes it easier to handle in the list component

File: client/packages/coldchain/src/Equipment/api/api.ts [hunk 2]
```diff
@@ -83,12 +93,21 @@ export const getAssetQueries = (sdk: Sdk, storeId: string) => ({
...
+
+      const items = result?.assetLogs;
+
       return items;
     },
   },
   insert: async (input: InsertAssetInput): Promise<string> => {
     const result = await sdk.insertAsset({
-      input,
+      input: { ...input, storeId },
```
Comment (mark-prins) [inline client/packages/coldchain/src/Equipment/api/api.ts:110]: leftover task : the asset should be store specific, we'll do the multi store version on central.

**Review by jmbrunskill — APPROVED** [review]
Looks good to me.

A couple of comments, looks like we still have to refresh to update the asset status on the summary page after entering a new status?

We don't seem to be logging the asset details on CREATE, but are for UPDATE, seems a bit inconsistent? Maybe not a big deal though...
image

**Comment by mark-prins** [comment]
retested the status update - and working for me (see below) so I'll merge and monitor 🙏 


https://github.com/msupply-foundation/open-msupply/assets/9192912/9c82a398-d9a4-400a-ad6c-1883781d873e

**Pull request merged**